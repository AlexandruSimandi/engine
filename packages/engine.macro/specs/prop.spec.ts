import pluginTester from "babel-plugin-tester";
import prettier from "prettier";
import plugin from "babel-plugin-macros";

const macroFile = "'@c11/engine.macro'";
const config = require("./babelConfig.json");

pluginTester({
  plugin,
  babelOptions: { filename: __filename, ...config },
  formatResult: (result: any) => {
    return prettier.format(result, {
      parser: "babel",
    });
  },
  tests: {
    "should support Prop": {
      code: `
        import { producer } from ${macroFile}
        const result: producer = ({
          foo = Prop.foo,
          bar = Prop.foo[Prop.baz],
          baz = Prop.foo[Arg.bar][Prop.baz]
        }) => { }
      `,
      snapshot: true,
    },
  },
});
