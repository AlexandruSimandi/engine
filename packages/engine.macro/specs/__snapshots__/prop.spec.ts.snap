// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`macros should support Prop: should support Prop 1`] = `

import { producer } from '@c11/engine.macro'
const result: producer = ({
  foo = Prop.foo,
  bar = Prop.foo[Prop.baz],
  baz = Prop.foo[Arg.bar][Prop.baz]
}) => { }

      ↓ ↓ ↓ ↓ ↓ ↓

"use strict";

var result = {
  args: {
    type: "STRUCT",
    value: {
      foo: {
        type: "VALUE",
        value: {
          type: "EXTERNAL",
          path: ["foo"],
        },
      },
      bar: {
        type: "VALUE",
        value: {
          type: "EXTERNAL",
          path: ["foo", "@baz"],
        },
      },
      baz: {
        type: "VALUE",
        value: {
          type: "EXTERNAL",
          path: ["foo", "$bar", "@baz"],
        },
      },
    },
    meta: {},
  },
  fn: function fn(_ref) {
    var foo = _ref.foo,
      bar = _ref.bar,
      baz = _ref.baz;
  },
};


`;
